src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1704,7 +1704,12 @@ void Graph::EnforceInferencePrecision() {
         if (nodesToSkip.count(node) && !node->enforceBF16evenForGraphTail)
             continue;
 
-        if (one_of(node->getType(), Type::Input, Type::Output, Type::MemoryInput, Type::MemoryOutput))
+        // Skip BF16 enforcement for the 'If' node.
+        // Because "If" contains two subgraphs. Whether any of nodes in subgraph is enforced to bf16 or not, the
+        // precision of input & output of the subgraph will be unchanged, so the input and output of "If" node itself
```

Comment:
BTW, why precisions of subgraph inputs/outputs are unchanged? In case of BF16 enforcement I would expect them to be BF16 to avoid unnecessary data_type conversions between main graph and subgraph.

Reply:
This is a good question and also is my concern.
1. Subgraph itself also will call `CreateGraph`, from `Graph::EnforceInferencePrecision()`, https://github.com/openvinotoolkit/openvino/blob/ea1ffbaf7f6eb9f343837ac7a6dc3e4774f4cd71/src/plugins/intel_cpu/src/graph.cpp#L1661, we can find: only the calculation nodes will be enforced with BF16.

2. https://github.com/openvinotoolkit/openvino/blob/ea1ffbaf7f6eb9f343837ac7a6dc3e4774f4cd71/src/plugins/intel_cpu/src/graph.cpp#L1707, The input/output nodes will be skipped(Means default precision will be adopted).

Reply:
It seems like the I/O nodes' precision must remain default, if we change I/O node precision, the architecture of inference must be updated, right? @usstq @dmitry-gorokhov

Reply:
Keep Input/Output precision is reasonable for main graph, since these precisions must consistent with user's expectation, but for subgraph it's changeable in theory, but it requires more works since it's a framework level adjustment.